hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/utils/KeyActivationUtils.java

```diff
@@ -69,14 +72,20 @@ private KeyActivationUtils() {
 	 * @return whether the implied key is active
 	 */
 	public static boolean validateKey(
-			final MessageFrame frame,
+			final InfoProvider provider,
 			final Address target,
 			final KeyActivationTest activationTest,
 			final WorldLedgers ledgers,
-			final ContractAliases aliases
+			final Optional<ContractAliases> aliases
 	) {
-		final var recipient = aliases.resolveForEvm(frame.getRecipientAddress());
-		final var sender = aliases.resolveForEvm(frame.getSenderAddress());
+		//logic for direct token account calls
+		if(provider instanceof DirectCallsInfoProvider){
+			final var message =((DirectCallsInfoProvider) provider).precompileMessage();
+			return activationTest.apply(false, target, message.getSenderAddress(), ledgers);
+		}
+		final var frame = ((EVMInfoProvider)provider).messageFrame();
+		final var recipient = aliases.get().resolveForEvm(frame.getRecipientAddress());
+		final var sender = aliases.get().resolveForEvm(frame.getSenderAddress());
 
```

Comment:
There might be a case (even if not present now) where we can have an `EVMInfoProvider` but have not set `ContractAliases` optional value. In this case we would have `NullPointerException`. We can think of a graceful way to handle this case.